### PR TITLE
[CE-953] Support Cirrus CI

### PR DESCRIPTION
**Pull request by fkorotkov**
Support reporting from Cirrus CI

### CODE REVIEW

**Comment by fkorotkov** [comment]
BTW master CI is broken. Will try to investigate it.

File: tests/test [hunk 1]
```diff
@@ -399,6 +399,19 @@ function test_shippable (){
...
+    reset
+    export CIRRUS_CI="true"
+    export CIRRUS_REPO_FULL_NAME='codecov/ci-repo'
+    export CIRRUS_BRANCH="master"
+    export CIRRUS_PR="1"
+    export CIRRUS_CHANGE_IN_REPO="c739768fcac68144a3a6d82305b9c4106934d31a"
+    export CIRRUS_BUILD_ID="777"
+    export CIRRUS_TASK_ID="239"
+    export CIRRUS_TASK_NAME="test"
+    assertURL "https://codecov.io/upload/v4?package=bash-tbd&token=&branch=master&commit=c739768fcac68144a3a6d82305b9c4106934d31a&build=239&build_url=&name=&tag=&slug=codecov%2Fci-repo&service=cirrus&flags=&pr=1&job=test"
```
Comment (AlexWayfer) [inline tests/test]: You have `TASK_ID` as `build`, `TASK_NAME` as `job` and no `BUILD_ID` in URL. Are you sure that this is correct?

Comment (fkorotkov) [reply tests/test]: In Cirrus CI you have a build which is a collection of tasks that are reported independently. And `codecov` is reported from a single task so I think using `TASK_ID` and `TASK_NAME` is accurate. Maybe URL should point to the task as well for consistency. 🤔

Comment (AlexWayfer) [reply tests/test]: I mean… you're stubbing unused `CIRRUS_BUILD_ID`.

**Comment by fkorotkov** [comment]
Regarding failing CI I'm giving up. There are 3 failures left that I cannot understand in #141

**Comment by vanesyan** [comment]
Any updates?

**Comment by fkorotkov** [comment]
@vanesyan no updates, tried to understand the failures [once again](https://github.com/codecov/codecov-bash/pull/141#issuecomment-506827656) but without success. Pinged @stevepeak though, will see if he'll respond. 🤷‍♂

**Comment by fkorotkov** [comment]
BTW I believe CI failure is not related to the change since master is broken at the moment.

**Comment by drazisil** [comment]
RE-running CI to check if tests pass.

**Comment by drazisil** [comment]
Tests pass, we need to add this to the CI whitelist on the backend before I can merge.

Will keep you posted, thank you for the PR, @fkorotkov

**Comment by drazisil** [comment]
Backed PR created, will approve this, and update when deployed.

**Comment by fkorotkov** [comment]
Awesome! Thank you, @drazisil!

**Comment by drazisil** [comment]
This has been deployed, Thank you for your patience.

**Comment by drazisil** [comment]
This has been deployed, Thank you for your patience.

**Comment by AlexWayfer** [comment]
Stop, what does this addition mean? I thought I'll be able to use Cirrus CI + Codecov without tokens, like in Travis or CircleCI.

**Pull request merged**